app/controllers/forms_controller.rb

```diff
@@ -105,6 +105,8 @@ def odk_manifest
     authorize!(:download, @form)
     @cache_key = @form.odk_download_cache_key
     unless fragment_exist?(@cache_key)
+      questions = @form.visible_questionings.collect(&:question).select(&:audio_prompt_file_name)
```

Comment:
Always use `map` instead of `collect` please.

Also note that this is only working because `audio_prompt_file_name` (a string) is 'truthy'. But it doesn't read very cleanly. On first inspection it looks like you're assembling an array of strings but really it's an array of questions. Can you refactor for clarity?